ibis/backends/tests/test_generic.py

```diff
@@ -163,3 +164,26 @@ def test_case_where(backend, alltypes, df):
     expected['new_col'] = expected['new_col']
 
     backend.assert_frame_equal(result, expected)
+
+
+@pytest.mark.xfail_unsupported
+@pytest.mark.skip_backends(['postgres'])
+def test_select_filter_mutate(backend, alltypes, df):
+    t = alltypes
+    t = t.mutate(
+        float_col=ibis.case()
+        .when(t['bool_col'], t['float_col'])
+        .else_(np.nan)
+        .end()
+    )
+    t = t[t.columns]
+    t = t[~t['float_col'].isnan()]
+    t = t.mutate(float_col=t['float_col'].cast('int32'))
+    result = t.execute()
+
+    expected = df.copy()
+    expected['float_col'][~df['bool_col']] = None
+    expected = expected[~expected['float_col'].isna()]
```

Comment:
add a .copy() here (or will show a warning) or use assign on the next line.

Reply:
Updated. Changed the next time to assign.